Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Application/AtomToolsApplication.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <AtomToolsFramework/Communication/LocalServer.h>
+#include <AtomToolsFramework/Communication/LocalSocket.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Debug/TraceMessageBus.h>
+#include <AzCore/UserSettings/UserSettingsProvider.h>
+#include <AzFramework/Application/Application.h>
+#include <AzFramework/Asset/AssetSystemBus.h>
+#include <AzFramework/Logging/LogFile.h>
+#include <AzQtComponents/Application/AzQtApplication.h>
+#include <AzToolsFramework/API/AssetDatabaseBus.h>
+#include <AzToolsFramework/API/EditorPythonConsoleBus.h>
+#include <AzToolsFramework/Logger/TraceLogger.h>
+
+#include <QTimer>
+
+namespace AtomToolsFramework
+{
+    //!Base class for Atom tools to inherit from
+    class AtomToolsApplication
+        : public AzFramework::Application
+        , public AzQtComponents::AzQtApplication
+        , protected AzToolsFramework::AssetDatabase::AssetDatabaseRequestsBus::Handler
+        , protected AzFramework::AssetSystemStatusBus::Handler
+        , protected AzToolsFramework::EditorPythonConsoleNotificationBus::Handler
+        , protected AZ::Debug::TraceMessageBus::Handler
+        , protected AZ::UserSettingsOwnerRequestBus::Handler
+    {
+    public:
+        AZ_TYPE_INFO(AtomTools::AtomToolsApplication, "{30F90CA5-1253-49B5-8143-19CEE37E22BB}");
```

Comment:
This GUID is duplicated with some other classes. You can use VS2019 Tools -> Create GUID to create a new one. 
I noticed the ShaderManagementConsoleApplication and MaterialEditorApplication are using the same one. Do you mind to update ShaderManagementConsoleApplication's GUID too?